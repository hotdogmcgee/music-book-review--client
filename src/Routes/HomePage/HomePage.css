body {
    margin: 0
}

section {
    text-align: center;
    margin-bottom: 20px;
}



.select-boxes {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.select-box {
    height: 100px;
    width: 100px;
    border: 1px solid black;
    margin: 20px;
}