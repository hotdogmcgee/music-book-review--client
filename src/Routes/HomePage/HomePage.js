import React from "react";
import "./HomePage.css";
import { Section } from "../../Components/Utils/Utils";
import BookListContext from "../../Contexts/BookListContext";
import BookList from "../../Components/BookList/BookList";
import CardList from "../../Components/CardList/CardList";

class HomePage extends React.Component {
  state = {
    searchValue: ""
  };
  static contextType = BookListContext;




  makeInstrumentList = () => {
    const { bookList } = this.context
    let instrumentCount = {}

    const instruments = bookList.map(item => item.instrument)

    for (let i = 0; i < instruments.length; i++) {
      const num = instruments[i]
      instrumentCount[num] = instrumentCount[num] ? instrumentCount[num] + 1 : 1;

    }
    
    let numberOfBooks = Object.keys(instrumentCount).map(key => {
      return {
      title: key,
      numEntries: instrumentCount[key]   
    }
    })

    //returns list sorted by most books for a given instrument
    const sorted = numberOfBooks.sort((a, b) => { return b.numEntries - a.numEntries} )

    return sorted
    
  }
  renderBookListOrCardList = () => {

    const instrumentList =     this.makeInstrumentList()
    const searchValue = this.context.filterObject.searchValue || "";

    let newList = this.context.bookList;
    if (searchValue.length > 0) {
      newList = newList.filter(book => {
        const lc = book.title.toLowerCase();
        const filter = searchValue.toLowerCase();
        return lc.includes(filter);
      });
      return <BookList bookList={newList} />;
    } else {
      return (
        <div>
          <Section>
            <p className="HomePage__hero">
              The Music Book Reviews app helps you find the right book for a
              variety of instruments.
              <br></br>
              <br></br>
              Add your own reviews to grow the project!
            </p>
          </Section>
          <h2>Instrument select</h2>
          <CardList instruments={instrumentList} />
        </div>
      );
    }
  };

  render() {
    return (
      <section className="HomePage">
        <Section className="HomePage__display-area">
          {this.renderBookListOrCardList()}
        </Section>
      </section>
    );
  }
}

export default HomePage;
