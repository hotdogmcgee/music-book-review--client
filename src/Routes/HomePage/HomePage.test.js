import React from "react";
import HomePage from "./HomePage";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("HomePage component", () => {
  it("renders a HomePage by default", () => {
    const context = {
      bookList: [],
      filterObject: {
        searchValue: "",
        browseValue: "",
        instrumentValue: "",
        filterValue: "",
        sortValue: "",
        listSorted: false
      }
    };
    const wrapper = shallow(<HomePage />, { context });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
