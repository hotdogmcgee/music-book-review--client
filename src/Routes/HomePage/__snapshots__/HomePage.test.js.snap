// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders a HomePage by default 1`] = `
<section
  className="HomePage"
>
  <Section
    className="HomePage__display-area"
  >
    <div>
      <Section>
        <p
          className="HomePage__hero"
        >
          The Music Book Reviews app helps you find the right book for a variety of instruments.
          <br />
          <br />
          Add your own reviews to grow the project!
        </p>
      </Section>
      <Section
        className="HomePage__how-to"
      >
        <Button
          className="show-how-to-button"
          onClick={[Function]}
        >
          How do I use this app?
        </Button>
      </Section>
      <h2>
        Instrument select
      </h2>
      <CardList
        instruments={Array []}
      />
    </div>
  </Section>
</section>
`;
